internal/cmd/generate/commands/gensource/generator.go

```diff
@@ -749,6 +749,23 @@ func (r ` + g.Endpoint.MethodWithNamespace() + `Request) Do(ctx context.Context,
 		return nil, err
 	}` + "\n\n")
 
+	g.w(`if res.StatusCode != http.StatusOK {
```

Comment:
Initial note will need to update this to follow the "non 2xx" logic in `IsError`. 

Ref https://github.com/elastic/go-elasticsearch/blob/master/esapi/esapi.response.go#L74-L76

Likewise will need to figure out what to do with `IsError` since its incompatible with these changes, I suspect it could be removed entirely, but not sure how wide spread its usage may be.